src/ipc/ipc3/helper.c

```diff
@@ -572,6 +572,10 @@ static int ipc_comp_to_buffer_connect(struct ipc_comp_dev *comp,
 	tr_dbg(&ipc_tr, "ipc: comp sink %d, source %d -> connect", buffer->id,
 	       comp->id);
 
+	if (comp->core != buffer->cb->core) {
+		tr_err(&ipc_tr, "ipc: shared buffers are not supported for IPC3");
+		return -ENOTSUP;
+	}
```

Comment:
This is interesting. I think this might be ok given I don't think we have any mainline users who need this feature. And if in future somebody needs this, one option to create all buffers as shared (as a build-time option to support this type of usage).

Reply:
```
static int ipc_comp_to_buffer_connect(struct ipc_comp_dev *comp,
				      struct ipc_comp_dev *buffer)
{
	tr_dbg(&ipc_tr, "ipc: comp sink %d, source %d -> connect", buffer->id,
	       comp->id);
#if CONFIG_INCOHERENT
	if (comp->core != buffer->cb->core) {
		tr_err(&ipc_tr, "ipc: shared buffers are not supported for IPC3 on incoherent architectures");
		return -ENOTSUP;
	}
#endif
	return comp_buffer_connect(comp->cd, comp->core, buffer->cb,
				   PPL_CONN_DIR_COMP_TO_BUFFER);
}

```

I'm not very familiar with other architectures, should I make this change?

Reply:
line 457 are aligned with line 575-578, buffer is not shared and if core is not same, then return with error. 

if add CONFIG_INCOHERENT limitation, then line 457 need check also to determine allocate shared or not shared buffer.

Reply:
I think @marcinszkudlinski this would be fine. We don't have any INCOHERENT, multi-core boards supported in upstream other than the Intel platforms we've moved to IPC4, so this does not effect anyone at the moment, but codifiying this limitation in code like this would be good.

Reply:
@btian1 
>  then line 457 need check also to determine allocate shared or not shared buffer.

That's the whole point, at this moment there's no information about the modules the buffer will be bind to 
So we have we can either use all buffers as shared or limit usage

Reply:
```
	/* register buffer with pipeline */
#if CONFIG_USE_SHARED_BUFFERS_FOR_IPC3
	buffer = buffer_new(desc, true);
#else
	buffer = buffer_new(desc, false);
#endif
```
```
static int ipc_comp_to_buffer_connect(struct ipc_comp_dev *comp,
				      struct ipc_comp_dev *buffer)
{
	tr_dbg(&ipc_tr, "ipc: comp sink %d, source %d -> connect", buffer->id,
	       comp->id);

#if !CONFIG_USE_SHARED_BUFFERS_FOR_IPC3
	if (comp->core != buffer->cb->core) {
		tr_err(&ipc_tr, "ipc: shared buffers are not supported for IPC3");
		return -ENOTSUP;
	}
#endif
	return comp_buffer_connect(comp->cd, comp->core, buffer->cb,
				   PPL_CONN_DIR_COMP_TO_BUFFER);
}
```

Reply:
as discussed - it is an overkill, just stay with a check with if coherent